swift/ql/lib/codeql/swift/dataflow/internal/TaintTrackingPublic.qll

```diff
@@ -37,4 +37,9 @@ predicate defaultImplicitTaintRead(DataFlow::Node node, DataFlow::ContentSet cs)
     cx.asNominalTypeDecl() = d and
     cs.getAReadContent().(DataFlow::Content::FieldContent).getField() = cx.getAMember()
   )
+  or
+  // We often expect taint to reach a sink inside `CollectionContent`, for example an array element
+  // or pointer contents. It is convenient to have a default implicit read step for these cases rather
+  // than implementing this step in a lot of separate `allowImplicitRead`s.
+  cs.getAReadContent() instanceof DataFlow::Content::CollectionContent
```

Comment:
If we ever run into performance problems here a natural thing to do would be to add a conjunct that further restricts `node` (for example: it's of a type that can actually carry `CollectionContent`). But since we're not observing any performance issues here yet I don't think we need to go down that rabbit hole for now.

Reply:
Yeah, indeed Java already does this.  My worry is that we miss out some nodes that could hold collection content and thus end up missing results that we could have got for less effort.  Absolutely an option if we see any problems though.

The DCA run looks fine, aside from a bit of wobble.